SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesWithSpecifiedDataService.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using BookingsApi.Client;
+using BookingsApi.Contract.Responses;
+using Microsoft.Extensions.Logging;
+using SchedulerJobs.Common.Constants;
+using UserApi.Client;
+using UserApi.Contract.Responses;
+using VideoApi.Client;
+using VideoApi.Contract.Requests;
+
+namespace SchedulerJobs.Services
+{
+    public interface IAnonymiseHearingsConferencesWithSpecifiedDataService
+    {
+        Task AnonymiseHearingsConferencesWithSpecifiedData();
+    }
+
+    public class
+        AnonymiseHearingsConferencesWithSpecifiedDataService : IAnonymiseHearingsConferencesWithSpecifiedDataService
+    {
+        private readonly IVideoApiClient _videoApiClient;
+        private readonly IBookingsApiClient _bookingsApiClient;
+        private readonly IUserApiClient _userApiClient;
+        private readonly ILogger<AnonymiseHearingsConferencesWithSpecifiedDataService> _logger;
+
+        public AnonymiseHearingsConferencesWithSpecifiedDataService(IVideoApiClient videoApiClient,
+            IBookingsApiClient bookingsApiClient,
+            IUserApiClient userApiClient,
+            ILogger<AnonymiseHearingsConferencesWithSpecifiedDataService> logger)
+        {
+            _videoApiClient = videoApiClient;
+            _bookingsApiClient = bookingsApiClient;
+            _userApiClient = userApiClient;
+            _logger = logger;
+        }
+
+        public async Task AnonymiseHearingsConferencesWithSpecifiedData()
+        {
+            var anonymisationData = await _bookingsApiClient.GetAnonymisationDataAsync();
+
+            await _videoApiClient.AnonymiseConferenceWithHearingIdsAsync(new AnonymiseConferenceWithHearingIdsRequest
+                { HearingIds = anonymisationData.HearingIds });
+            await _videoApiClient.AnonymiseQuickLinkParticipantWithHearingIdsAsync(
+                new AnonymiseQuickLinkParticipantWithHearingIdsRequest { HearingIds = anonymisationData.HearingIds });
+
+            await _bookingsApiClient.AnonymiseParticipantAndCaseByHearingIdAsync("hearingIds",
+                anonymisationData.HearingIds);
+
+            if (anonymisationData.Usernames != null)
```

Comment:
Just a thought, could we invert this if statement to reduce nesting?

```
if (anonymisationData.Usernames == null)
{
  return;
}
            
foreach (var username in anonymisationData.Usernames)
{
  ...
}
```

Reply:
yea can do that, however I was wondering if there is something more I can do around this. Can explain why I have done it this way because the whole try catch here isnt ideal I am aware.